include/mbedtls/ssl_ciphersuites.h

```diff
@@ -315,6 +352,105 @@ mbedtls_sig_type_t mbedtls_ssl_get_ciphersuite_sig_alg( const mbedtls_ssl_cipher
 int mbedtls_ssl_ciphersuite_uses_ec( const mbedtls_ssl_ciphersuite_t *info );
 int mbedtls_ssl_ciphersuite_uses_psk( const mbedtls_ssl_ciphersuite_t *info );
 
+#if defined(MBEDTLS_KEY_EXCHANGE__SOME_PFS__ENABLED)
+static inline int mbedtls_ssl_ciphersuite_has_pfs( const mbedtls_ssl_ciphersuite_t *info )
```

Comment:
I think this could be merged with its `no_pfs()` sibling, and be declared unconditionally, as we always have at least one ciphersuite, and it always either supports PFS or not. Otherwise in the common case where both kinds of ciphersuites are defined, we end up with  essentially duplicated functions, wasting ROM (and perplexing programmers).

(If you really want to kill it, you can make the function always return 0 or always 1 when it's clear from what's not enabled that a switch statement is not needed.)